lib/iris/tests/test_merge.py

```diff
@@ -1098,5 +1099,48 @@ def test_splitattrs_keys_local_match_masks_global_mismatch(self):
         )
 
 
+@pytest.mark.parametrize(
+    "dtype", [np.int16, np.int32, np.int64, np.float32, np.float64]
+)
+class TestCubeMerge_masked_scalar:
+    """Test for merging of scalar coordinates containing masked data."""
+
+    def _build_cube(self, scalar_data):
+        return iris.cube.Cube(
+            np.arange(5),
+            standard_name="air_pressure",
+            aux_coords_and_dims=[
+                (AuxCoord(points=scalar_data, standard_name="realization"), None)
+            ],
+        )
+
+    def test_merge_scalar_coords_all_masked(self, dtype):
+        """Test merging of scalar aux coords all with masked data."""
+        n = 5
+        cubes = iris.cube.CubeList(
+            [self._build_cube(np.ma.masked_all(1, dtype=dtype)) for i in range(n)]
+        )
+        merged = cubes.merge_cube()
+        c = merged.coord("realization")
+        assert np.ma.isMaskedArray(c.points)
+        assert np.all(c.points.mask)
+        assert c.points.dtype.type is dtype
+
+    def test_merge_scalar_coords_some_masked(self, dtype):
+        """Test merging of scalar aux coords with mix of masked and unmasked data."""
+        n = 5
+        cubes = iris.cube.CubeList(
+            [
+                self._build_cube(np.ma.masked_array(i, dtype=dtype, mask=i % 2))
+                for i in range(n)
+            ]
+        )
+        merged = cubes.merge_cube()
+        c = merged.coord("realization")
+        assert np.ma.isMaskedArray(c.points)
+        assert np.any(c.points.mask) and not np.all(c.points.mask)
```

Comment:
There is enough information to assert that the original mask has been preserved.

```suggestion
        assert all([c.points.mask[i] == i % 2 for i in range(len(c.points))])
```

... or some less-ugly equivalent

Reply:
That's a good point. Updated with more rigorous test: e9551164e